src/NUnitFramework/tests/Attributes/TestCaseSourceTests.cs

```diff
@@ -290,6 +290,21 @@ public void HandlesExceptionInTestCaseSource_GuiDisplay(string lhs, string rhs)
             Assert.AreEqual(lhs, rhs);
         }
 
+
+        private static IEnumerable<TestCaseData> ZeroTestCasesSource() => Enumerable.Empty<TestCaseData>();
+
+        [TestCaseSource(nameof(ZeroTestCasesSource))]
```

Comment:
Well, if I'd only known that! Of course, we set this standard a long time back and it might not have been available. In any case, I agree with your logic. But since we'r making a real effort to create and empower a committee that will run NUnit in the future, I'd really like to wait for them to change the policy. Sorry for the trouble - it is in the coding standard, however.